internal/pkg/cacheutil/cacheserver.go

```diff
@@ -98,11 +102,10 @@ func (i IncomingDataCache) Put(key string) {
 	i.hosts[key] = NewShardedIncomingDataCache(i.maxTTL)
 }
 
-//GetHosts  Get All hosts
-func (i IncomingDataCache) GetHosts() map[string]*ShardedIncomingDataCache {
+//GetHosts  GetHosts locks the cache and returns the whole cache together with the lock
```

Comment:
It might be useful to include a comment here that the caller *must* unlock the semaphore when finished.

Reply:
nstead of calling lock.Unlock() use defer
defer lock.Unlock()
or
defer func(){
lock.Unlock()
}